plugins/woocommerce/src/Caches/OrderAggregateCacheService.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Caches;
+
+/**
+ * A service class to help with updates to the aggregate orders cache.
+ */
+class OrderAggregateCacheService {
+
+	/**
+	 * The orders cache to use.
+	 *
+	 * @var OrderCache
+	 */
+	private $order_aggregate_cache;
+
+	/**
+	 * Class initialization, invoked by the DI container.
+	 *
+	 * @internal
+	 * @param OrderAggregateCache $order_aggregate_cache The aggregate order cache engine to use.
+	 */
+	final public function init( OrderAggregateCache $order_aggregate_cache ) {
+		$this->order_aggregate_cache = $order_aggregate_cache;
+		add_action( 'woocommerce_new_order', array( $this, 'update_on_new_order' ) );
```

Comment:
We should probably consider `woocommerce_delete_order` as well.

Reply:
This has been added.